adding common actions for maintenance (#203)
OHS-Hosting-Infrastructure/infrastructure: docs/runbooks/maintenance.md

# Maintenance

Common things to note/do during maintenance.

## Add Disk Space

> 1. Create a new volume from the AWS console.

Reviewer: ```suggestion
1. Create a new volume from the AWS console Under Elastic Block Store > Volumes.
```

> 1. Attatch the volume to the instance from the AWS console.

Reviewer: ```suggestion
1. Attach the volume to the EC2 instance you are adding disk space to from the AWS console.
```

> 1. find the new device in /dev/

Reviewer: ```suggestion
1. SSH onto the EC2 instance you attached the new volume to and find the new device in the `/dev/` directory.
```

Also--how does one id the new volume in `dev`? The things I think are volumes (e.g. `xvdb`) don't have anything that I recognize from what's available in the AWS console.

> 1. Attach the volume to the EC2 instance you are adding disk space to from the AWS console.

Reviewer: ```suggestion
1. Attach the volume to the EC2 instance you are adding disk space to from the AWS console. Note: The naming convention is different between AWS and what you'll see on disk. What we've found is that on disk, `xv` corresponds to `s`. So in the console you'll see, for example, `sdf` and on disk you'll see `xvdf1`.
```

1. Partition the new disk device. These instructions suggest `fdisk` but `parted` is a good utility to use as well.

>    1. Start the `fdisk` utility You should *not* have the disk you are editing with `fdisk` mounted.

Reviewer: ```suggestion
   1. Start the `fdisk` utility. You should *not* have the disk you are editing with `fdisk` mounted.
```

I don't know what this means: "You should *not* have the disk you are editing with `fdisk` mounted."
Should I be writing `fdisk [options] <disk> ` to change the partition table?

   1. `p` to print the partition table as is

>    1. `n` to create a new partition.

Reviewer: These two options don't appear for me in fdisk

```
[centos@mariadb dev]$ fdisk
Usage:
 fdisk [options] <disk>    change partition table
 fdisk [options] -l <disk> list partition table(s)
 fdisk -s <partition>      give partition size(s) in blocks

Options:
 -b <size>             sector size (512, 1024, 2048 or 4096)
 -c[=<mode>]           compatible mode: 'dos' or 'nondos' (default)
 -h                    print this help text
 -u[=<unit>]           display units: 'cylinders' or 'sectors' (default)
 -v                    print program version
 -C <number>           specify the number of cylinders
 -H <number>           specify the number of heads
 -S <number>           specify the number of sectors per track
```

Author: i thought that should drop you into the utility to manage the disk

Reviewer: Thanks for this change. So I thought the device id would be something like `xvdd1` in MariaDB instance. However, when I run 
```
[centos@mariadb ~]$ fdisk /dev/xxdd1
fdisk: cannot open /dev/xxdd1: No such file or directory
```
What is the device id?

Reviewer: Also, is "the disk you are editing with" the one that you're sshing into? I don't understand the differentiation you are making there.

   1. see [fdisk documentation](https://tldp.org/HOWTO/Partition/fdisk_partitioning.html) if additional formatting is needed.

> 1. Add a filesystem on top.

Reviewer: ```suggestion
1. Add a `ext4` filesystem on top.
```

Is this a good guide on how to do: https://opensource.com/article/19/4/create-filesystem-linux-partition?

   1. use `mkfs.ext4` on the partition device you created.

> 1. Add a label to the new device to identify it.

Reviewer: Can you suggest a convention for the label?

> 1. Add new device mount using the label to /etc/fstab so that it will automatically mounted on a reboot.

Reviewer: Is this a good guide of how to do: https://linuxhint.com/mount_partition_uuid_label_linux/ ?

## Update NodeJS to version 12

Check the version of node installed:

`node --version`

Clean the yum caches:

`yum clean all`

Add the NodeSource yum repo and review the script to do so [here](https://rpm.nodesource.com/setup_12.x).

`curl -sL https://rpm.nodesource.com/setup_12.x | bash -`

Check that you see new versions of node available:

`yum list available nodejs`

Remove the current version of nodejs:

`yum remove nodejs`

To install the latest available nodejs from nodesource:

`yum install nodejs`

## Updating packages on Ubuntu

Check the available versions of a package:

`apt list -a <package name>`

To install the latest version of a package available:

`apt-get install <package name>`

To install at a particular version of a package:

> `apt-get install <package name>=<version>`

Reviewer: Did we decide to avoid adding repos/doing upgrades? Or is this worth adding?
https://ubiq.co/tech-blog/upgrade-apache-version-ubuntu/

Author: adding ppa's is not necessarly something I want us to go to. I'll add the link though.

## Add Disk Space

1. Create a new volume from the AWS console Under Elastic Block Store > Volumes.
1. Attach the volume to the EC2 instance you are adding disk space to from the AWS console. Note: The naming convention is different between AWS and what you'll see on disk. What we've found is that on disk, `xv` corresponds to `s`. So in the console you'll see, for example, `sdf` and on disk you'll see `xvdf`.
1. SSH onto the EC2 instance you attached the new volume to and find the new device in the `/dev/` directory. Confirm the name of the disk. In our example, `sdf` is `/dev/xvdf` but you may also see a `/dev/xvdf1` if the drive already had a partition.
1. Partition the new disk device. Note: These instructions suggest `fdisk` but `parted` is a good utility to use as well.
   1. Start the `fdisk` utility by running `fdisk /dev/<DEVICE ID>`. In our example, this would be `/dev/xvdd`. You should *not* have the disk you are editing with `fdisk` mounted.
   1. `p` to print the partition table as is
   1. `n` to create a new partition.
   1. see [fdisk documentation](https://tldp.org/HOWTO/Partition/fdisk_partitioning.html) if additional formatting is needed.

> 1. Add a `ext4` filesystem on top. Note, you may want an xfs file system but that will use different utlities.

Reviewer: Can you include why one chooses one file system over another? i.e. Why might you want an xfs? Is it just that you should check to see whatever the filesystem is for the disk you're enlarging?
